tests/cpp_unit_tests/test_dataset.cpp

```diff
@@ -115,9 +174,161 @@ DatasetType create_dataset(bool const is_batch, Idx const batch_size, MetaDatase
     CHECK(info.component_info.empty());
     return dataset;
 };
+
+void check_nan_or_equal(double first, double second) {
+    CHECK(((first == second) || (is_nan(first) == is_nan(second))));
+}
+
+void check_nan(double value) { CHECK(is_nan(value)); }
```

Comment:
This function can be directly used

Reply:
well, yes but it looks nicer this way because otherwise, when it's used, there's a mixture of `CHECK(foo(bar))` and `check_foo(bar)`. I prefer to have them all be `CHECK` or `check_`, but not a mixture. Optional though

Reply:
I will keep `check_nan` because of Martjin's reason above. Since the `CHECK`s are done through auxiliary functions like this one, even though redundant, the same style/convention is mantained througout the tests.